src/server/mod.rs

```diff
@@ -1664,6 +1666,189 @@ impl LanguageServer for LspServer {
                 }
                 Ok(None)
             }
+            Ok(LspServerCommand::CompositionInitialize) => {
+                let command_params: CompositionInitParams =
+                    match serde_json::value::from_value(
+                        params.arguments[0].clone(),
+                    ) {
+                        Ok(value) => value,
+                        Err(err) => {
+                            return Err(LspError::InternalError(
+                                format!("{:?}", err),
+                            )
+                            .into())
+                        }
+                    };
+
+                let new_text = match self.state.lock() {
+                    Ok(mut state) => {
+                        if state.composition.is_none() {
+                            let file = self.store.get_ast_file(
+                                &command_params.text_document.uri,
+                            )?;
+                            state.composition = Some(RefCell::new(
+                                composition::Composition::new(file),
```

Comment:
Note to reviewers:
* the LSP composition.initialize() acts as a reset. Where the user can re-create starting with a new bucket.
* this is why we don't error if `state.composition.is_some()`. Instead, we only handle the is_none() case.

Reply:
no longer relevant.